test/unit/train.test.js

```diff
@@ -0,0 +1,79 @@
+import { createStore } from "redux";
+import train from "../../src/train";
+import { ResultsGrades } from "../../src/constants.js";
+import { ColumnTypes } from "../../src/constants.js";
+import rootReducer, {
+  setFirehoseMetricsLogger,
+  setInstructionsKeyCallback,
+  setImportedData,
+  setLabelColumn,
+  addSelectedFeature,
+  setColumnsByDataType,
+  setTestData,
+  getConvertedPredictedLabel
+} from "../../src/redux";
+
+describe("train functions", () => {
+  test("train and predict with numerical data", async () => {
+    const store = createStore(rootReducer);
+
+    store.dispatch(setFirehoseMetricsLogger(() => {}));
+    store.dispatch(setInstructionsKeyCallback(() => {}));
+
+    const data = [
+      { temperature: "0", cost: "20", rain: "1000" },
+      { temperature: "50", cost: "25", rain: "1010" },
+      { temperature: "100", cost: "30", rain: "1020" },
+      { temperature: "150", cost: "35", rain: "1030" }
+    ];
+
+    store.dispatch(setImportedData(data, false));
+    store.dispatch(setLabelColumn("cost"));
+    store.dispatch(addSelectedFeature("temperature"));
+    store.dispatch(addSelectedFeature("rain"));
+
+    train.init(store);
+    train.onClickTrain(store);
+
+    store.dispatch(setTestData({temperature: "150", rain: "1030"}));
+
+    train.onClickPredict(store);
+
+    const predictedValue = getConvertedPredictedLabel(store.getState());
+    expect(predictedValue).toBe(30);
```

Comment:
Hmmm. Looking at the training data, I'd expect the predictedValue to be 35 since the most similar training data point where temp is 150 and rain is 1030 is 35, not 30.  However, I do think it's valuable to have coverage while we figure out how to handle this situation, and getting this test set up merged in will allow us to iterate more quickly. Could we maybe include a comment about how if the K value optimization goes sideways with really small datasets such as this sample one?

Reply:
Oh, actually this makes sense, because the model is never trained on ` { temperature: "150", cost: "35", rain: "1030" }` because we pull that row to use as test data to calculate accuracy.

Reply:
Is there a way to set up these tests so we pass the training data rather than the whole dataset?

Reply:
Good question.  This is something of an architecture question, and comes down to boundaries between modules.  I chose to unit test the exports from `train.js`.  We could unit test at a deeper level, or break apart some of the steps encapsulated in `init`, but I think we should do that refactoring separately.  This is a good example of how API designs can be informed by unit testing, and vice versa.